### ISSUE TITLE
Data size is wrong for cubemaps

### ISSUE THREAD

**Issue by expenses** [SiegeEngine/ddsfile#12]
I've been working on a ktx2 to dds converter, mostly because [renderdoc](https://renderdoc.org/) works well as a dds viewer but doesn't handle ktx2 files currently. Currently the size of `data` in `Dds` structs files created with `Dds::new_dxgi` and `is_cubemap` set to `true` is only the size of one of the cubemap faces.

Luckily, the `data` field is public so you can manipulate it without using any crate functions. Here's what I'm doing to pub a ktx2 cubemap in the right format:

```rust
let face_count = header.face_count as usize;

let mut faces = vec![Vec::new(); face_count];

for level in ktx2.levels() {
    let size = level.bytes.len() / face_count;
    for (i, face) in faces.iter_mut().enumerate() {
        let slice = &level.bytes[i * size..(i + 1) * size];
        face.extend_from_slice(slice);
    }
}

dds.data = faces.concat();
```

It'd be cool to have a really flexible api for manipulating the dds data. I'm not 100% sure what that'd be like yet though.

**Comment by mikedilger** [comment]
Thanks for that info.  I don't have time to maintain this crate that I wrote four years ago and don't myself use.  But if you provide a PR I'll merge it, or if you want to take over let me know.

**Comment by cwfitzgerald** [comment]
@mikedilger If you're looking for someone to maintain this, I'm currently working on a [compressed texture compression tool](https://github.com/cwfitzgerald/ctt) and am the maintainer of the [ktx2 crate](https://github.com/BVE-Reborn/ktx2) and would be happy to either take over, or help with maintenance of this crate.

**Comment by mikedilger** [comment]
> [@mikedilger](https://github.com/mikedilger) If you're looking for someone to maintain this, I'm currently working on a [compressed texture compression tool](https://github.com/cwfitzgerald/ctt) and am the maintainer of the [ktx2 crate](https://github.com/BVE-Reborn/ktx2) and would be happy to either take over, or help with maintenance of this crate.

Yes that would be great.  The source is also at PistonDevelopers/ddsfile.  SiegeEngine is pretty much dead.  If you make a clone and use that as the repo, we can somehow transfer crates.io ownership over to you (haven't done that in a while so I don't remember how it works).

**Comment by cwfitzgerald** [comment]
Sounds good! I'll get that fork going and make an issue pointing people that direction. 

As for crates.io, once you log in with GitHub, go to the crate, go to settings, you can add my GitHub username `cwfitzgerald` and once I accept it you can choose if you want to remove yourself or not.
